source/slang/slang-compiler.cpp

```diff
@@ -2255,6 +2262,29 @@ SlangResult dissassembleDXILUsingDXC(
                 SLANG_RETURN_ON_FAIL(writer.write(irModule, sourceManager, optionFlags, &serialData));
                 SLANG_RETURN_ON_FAIL(IRSerialWriter::writeContainer(serialData, compressionType, &container));
             }
+
+            auto entryPointCount = program->getEntryPointCount();
+            for( Index ii = 0; ii < entryPointCount; ++ii )
+            {
+                auto entryPoint = program->getEntryPoint(ii);
+                auto entryPointMangledName = program->getEntryPointMangledName(ii);
+
+                RiffContainer::ScopeChunk entryPointScope(&container, RiffContainer::Chunk::Kind::Data, IRSerialBinary::kEntryPointFourCc);
+
+                auto writeString = [&](String const& str)
```

Comment:
This doesn't take into account any alignment. As an implementation detail that will work currently as every write becomes a new data block which is ptr aligned, but that wasn't the intent. On reading if read on the helper is used, it uses memcpy so again all seems ok, but currently the helper assumes alignment. We could either make this code take into account alignment, or weaken the assert in helper.